server/server-storage/src/fallback.ts

```diff
@@ -0,0 +1,268 @@
+import {
+  withContext,
+  type Blob,
+  type MeasureContext,
+  type StorageIterator,
+  type WorkspaceId
+} from '@hcengineering/core'
+import { type Readable } from 'stream'
+
+import { getMetadata } from '@hcengineering/platform'
+import {
+  type BlobStorageIterator,
+  type BucketInfo,
+  type ListBlobResult,
+  type StorageAdapter,
+  type StorageAdapterEx,
+  type UploadedObjectInfo
+} from '@hcengineering/storage'
+
+import { Analytics } from '@hcengineering/analytics'
+import serverCore, { type StorageConfig, type StorageConfiguration } from '@hcengineering/server-core'
+
+class NoSuchKeyError extends Error {
+  code: string
+  constructor (msg: string) {
+    super(msg)
+    this.code = 'NoSuchKey'
+  }
+}
+
+/**
+ * Perform operations on storage adapter and map required information into BinaryDocument into provided DbAdapter storage.
+ */
+export class FallbackStorageAdapter implements StorageAdapter, StorageAdapterEx {
+  // Adapters should be in reverse order, first one is target one, and next ones are for fallback
+  constructor (readonly adapters: [string, StorageAdapter][]) {}
```

Comment:
can we use an object instead of [string, StorageAdapter] array? It is difficult to read code like `exAdapter.adapters[0][1]` :)

Reply:
Done.